tests/pipeline/test_engine.py

```diff
@@ -1562,3 +1567,43 @@ def test_no_groupby_maximum(self):
                         .reset_index(level=1, drop=True))
 
         assert_equal(groupby_max, pipeline_max)
+
+
+class ResolveDomainTestCase(zf.ZiplineTestCase):
+
+    def test_resolve_domain(self):
+        engine_generic = SimplePipelineEngine(
+            None, None, default_domain=GENERIC
+        )
+        engine_jp = SimplePipelineEngine(
+            None, None, default_domain=JP_EQUITIES
+        )
+
+        pipe_generic = Pipeline()
+        pipe_us = Pipeline(domain=US_EQUITIES)
```

Comment:
One more case worth testing here would be a pipeline with no explicit domain, but with a US-specialized column, to make sure that we're properly using the pipeline's inference.

Reply:
added